lib/Doctrine/ORM/Query/Expr/Join.php

```diff
@@ -104,7 +104,7 @@ public function getConditionType()
     }
 
     /**
-     * @return string
+     * @return string|Comparison|Composite
```

Comment:
```suggestion
     * @return string|Comparison|Composite|null
```

Seems like nullability is not documented correctly on any of the getters. Or am I missing something?

Reply:
Yes, that seems right IMO.
I didn't include this type because the same situation is present in other properties (`$alias`, `$conditionType`, `$indexBy`).
I think we should make a separate PR for these cases, as I suspect this is not the only class affected by this issue.
WDYT?